.github/workflows/ci-lint.yml

```diff
@@ -7,11 +7,28 @@ jobs:
     uses: ./.github/workflows/proto.yml
   golangci-lint:
     runs-on: ubuntu-latest
+    permissions:
+      checks: write # allow write access to checks to allow the action to annotate code in the PR.
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with: { fetch-depth: 0 }
+      - name: Load Version
+        id: load-version
+        run: |
+          echo "GOLANGCI_VERSION=v$(cat .golangci-version)"
       - name: golangci-lint
-        uses: reviewdog/action-golangci-lint@v2
+        uses: golangci/golangci-lint-action@v6
         with:
-          github_token: ${{ secrets.github_token }}
-          reporter: github-pr-review
-          golangci_lint_flags: --timeout 10m
+          version: ${{ steps.load-version.outputs.GOLANGCI_VERSION }}
+          args: -v -c .golangci.yml
+  golang-sec:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with: { fetch-depth: 0 }
+      - name: Run Gosec Security Scanner
+        uses: cosmos/gosec@v0.10.4
+        with:
+          args: ./...
```

Comment:
this should be added in a separate pr, and in a way that will not result in the master branch CI failing (make it only operate on a diff, disable rules that we break to be re-enabled one at a time in subsequent prs, or update code if it is very simple and obviously incorrect). 

if we break a rule that requires a meaningful & consequential change to the code, it should be made in a pr that highlights what is happening and why the change is safe.

on principle & to the best of our ability, we should never merge a commit that fails the PR CI or will break master's CI